### ISSUE TITLE
Embedded python 3.8 interpreter crashes after one evaluation

### ISSUE THREAD

**Issue by prudhomm** [pybind/pybind11#2153]
multiple(well 2) embedded python interpreter calls crash with pytbind11+python 3.8. 
The first works just fine, the second crashes.

here is a reproducer:

```c++
#include <pybind11/embed.h>
#include <sstream>

namespace py = pybind11;
using namespace py::literals;

int main(int argc, char** argv ) 
{
    for( int i = 0; i < 2; ++i )
    {
        py::scoped_interpreter guard{};

        auto locals = py::dict("f"_a="[x**2,y**2,z**2]");
        std::ostringstream ostr;
        ostr <<  "from sympy import *\n"
             << "x,y,z = symbols('x y z');"
             << "f=x**2;\n"
             << "print('f=', integrate(f,(x,0,1)))\n";
        py::exec(ostr.str().c_str(), py::globals(), locals);
    }
}
```
`py:eval` and `py::eval_file` fail similarly. 
It used to work fine with the previous 3.x versions.

I used an 'old' version of pybind11 as well as HEAD to check the crash.
am I missing something?

here is what the debugger says:
<img width="1214" alt="Screenshot 2020-04-06 23 18 23" src="https://user-images.githubusercontent.com/151693/78607560-9a0d2380-785f-11ea-9041-7c089db58279.png">

**Comment by carlsonmark** [comment]
Which 3.x versions did it work for for you?
This is a similar report: https://github.com/pybind/pybind11/issues/2069
I am not sure using `py::scoped_interpreter` more than once in an application is wise, since some Python libraries will not expect that to occur, and will not cleanup after themselves well.

My comment on 2069:

> Is it possible for you to only start the interpreter once in your application?
> 
> It could be the case that numpy can only be initialized once, but I am not sure.
> 
> Some relevant documentation:
> 
> > Modules created with pybind11 can be safely re-initialized after the interpreter has been restarted. However, this may not apply to third-party extension modules. The issue is that Python itself cannot completely unload extension modules and there are several caveats with regard to interpreter restarting. In short, not all memory may be freed, either due to Python reference cycles or user-created global data. All the details can be found in the CPython documentation.

**Comment by prudhomm** [comment]
3.7 worked perfectly fine and earlier also. In most apps we use the interpreter only once so far but that may change. I can certainly change the behavior of our apps. It seems indeed a duplicate of #2069. I thought I did my homework. Actually the issue started at 3.8 for us.

**Comment by carlsonmark** [comment]
Interesting. I don't think it worked even with 3.5 for me, using old versions of numpy/scipy. I think it is more dependant on the versions of the third-party libraries being used than it is on the Python version in our case.

**Comment by prudhomm** [comment]
I just removed all scoped_interpreter instances and have now initialize_interpreter called at the very start . It is all good now.
should I keep this open or should I close it ?

**Comment by carlsonmark** [comment]
I'd say close it. I don't think there is anything pybind11 can do about it. If that changes, then it can be reopened (or another one can be created.)

**Comment by prudhomm** [comment]
and thanks a lot @carlsonmark  for the nice discussion!
